pkg/util/utils.go

```diff
@@ -71,118 +73,175 @@ func StringInSlice(s string, sl []string) bool {
 	return false
 }
 
-// ParseChanges returns key, value(s) pair for given option.
-func ParseChanges(option string) (key string, vals []string, err error) {
-	// Supported format as below
-	// 1. key=value
-	// 2. key value
-	// 3. key ["value","value1"]
-	if strings.Contains(option, " ") {
-		// This handles 2 & 3 conditions.
-		var val string
-		tokens := strings.SplitAfterN(option, " ", 2)
-		if len(tokens) < 2 {
-			return "", []string{}, fmt.Errorf("invalid key value %s", option)
-		}
-		key = strings.Trim(tokens[0], " ") // Need to trim whitespace part of delimiter.
-		val = tokens[1]
-		if strings.Contains(tokens[1], "[") && strings.Contains(tokens[1], "]") {
-			//Trim '[',']' if exist.
-			val = strings.TrimLeft(strings.TrimRight(tokens[1], "]"), "[")
-		}
-		vals = strings.Split(val, ",")
-	} else if strings.Contains(option, "=") {
-		// handles condition 1.
-		tokens := strings.Split(option, "=")
-		key = tokens[0]
-		vals = tokens[1:]
-	} else {
-		// either ` ` or `=` must be provided after command
-		return "", []string{}, fmt.Errorf("invalid format %s", option)
-	}
-
-	if len(vals) == 0 {
-		return "", []string{}, errors.Errorf("no value given for instruction %q", key)
-	}
-
-	for _, v := range vals {
-		//each option must not have ' '., `[`` or `]` & empty strings
-		whitespaces := regexp.MustCompile(`[\[\s\]]`)
-		if whitespaces.MatchString(v) || len(v) == 0 {
-			return "", []string{}, fmt.Errorf("invalid value %s", v)
-		}
-	}
-	return key, vals, nil
-}
-
-// GetImageConfig converts the --change flag values in the format "CMD=/bin/bash USER=example"
-// to a type v1.ImageConfig
+// GetImageConfig produces a v1.ImageConfig from the --change flag that is
+// accepted by several Podman commands. It accepts a (limited subset) of
+// Dockerfile instructions.
 func GetImageConfig(changes []string) (v1.ImageConfig, error) {
-	// USER=value | EXPOSE=value | ENV=value | ENTRYPOINT=value |
-	// CMD=value | VOLUME=value | WORKDIR=value | LABEL=key=value | STOPSIGNAL=value
-
-	var (
-		user       string
-		env        []string
-		entrypoint []string
-		cmd        []string
-		workingDir string
-		stopSignal string
-	)
-
-	exposedPorts := make(map[string]struct{})
-	volumes := make(map[string]struct{})
-	labels := make(map[string]string)
-	for _, ch := range changes {
-		key, vals, err := ParseChanges(ch)
-		if err != nil {
-			return v1.ImageConfig{}, err
+	// Valid changes:
+	// USER
+	// EXPOSE
+	// ENV
+	// ENTRYPOINT
+	// CMD
+	// VOLUME
+	// WORKDIR
+	// LABEL
+	// STOPSIGNAL
+
+	config := v1.ImageConfig{}
+
+	for _, change := range changes {
+		// First, let's assume proper Dockerfile format - space
+		// separator between instruction and value
+		split := strings.SplitN(change, " ", 2)
+
+		if len(split) != 2 {
+			split = strings.SplitN(change, "=", 2)
```

Comment:
I'm debating dropping this... It'll break out test suite and anyone who used our old format, but it also makes parsing so much harder than it needs to be.